tests/test_pre_processing.py

```diff
@@ -1,14 +1,317 @@
+        Tests that a value error is raised when the annuity is left empty and not all other cost
+        parameters are included.
+        """
+        # copy scenario csv file to the package path
+        f_name = "annuity_empty_partial_cost_params.csv"
+        shutil.copy(os.path.join(self.test_inputs_pre_p, f_name), os.path.join(self.package_path, f_name))
+        # check if calling the pre_processing function with the given scenario raises a value error
+        with pytest.raises(ValueError):
+            pre_processing(self.pre_p_dir, wacc=1)
+
+    def test_annuity_defined_partial_cost_params_raises_error(self):
+        """
+        Tests that a value error is raised when the annuity is defined and some but not all other cost
+        parameters are included.
+        """
+        # copy scenario csv file to the package path
+        f_name = "annuity_defined_partial_cost_params.csv"
+        shutil.copy(os.path.join(self.test_inputs_pre_p, f_name), os.path.join(self.package_path, f_name))
+        # check if calling the pre_processing function with the given scenario raises a value error
+        with pytest.raises(ValueError):
+            pre_processing(self.pre_p_dir, wacc=1)
+
+    def test_annuity_empty_all_cost_params_defined_calculates_annuity(self):
+        """
+        Tests that the annuity is calculated and added to 'capacity_cost' when the annuity is left
+        empty and all other cost parameters are defined.
+        """
+        # copy scenario csv file to the package path
+        f_name = "annuity_empty_all_cost_params_defined.csv"
+        shutil.copy(os.path.join(self.test_inputs_pre_p, f_name), os.path.join(self.package_path, f_name))
+        # call the pre_processing function with wacc = 1
+        wacc = 1
+        pre_processing(self.pre_p_dir, wacc=wacc)
+        # check if the actual value matches the expected value
+        expected_value = calculate_annuity(capex=975000, opex_fix=11625, lifetime=20, wacc=wacc)
+        df = pd.read_csv(os.path.join(self.package_path, f_name), delimiter=';')
+        actual_value = df['capacity_cost'].iloc[0]
+        assert actual_value == expected_value
+
+    def test_annuity_empty_all_cost_params_defined_logs_message(self, caplog):
+        """
+        Tests that an info message is logged when a new annuity value is calculated and updated
+        from the other cost parameters.
+        """
+        # copy scenario csv file to the package path
+        f_name = "annuity_empty_all_cost_params_defined.csv"
+        shutil.copy(os.path.join(self.test_inputs_pre_p, f_name), os.path.join(self.package_path, f_name))
+        # check if the info message is logged when the pre_processing function is called
+        pre_processing(self.pre_p_dir, wacc=1)
+        assert any(record.levelname == "INFO" and "has been calculated and updated" in record.message
+                   for record in caplog.records)
+
+    def test_annuity_all_cost_params_some_empty_raises_error(self):
```

Comment:
If the annuity is provided, here as well I would only raise a warning, unless those parameters are needed for post-processing